### ISSUE TITLE
Unable to replace a placeholder with multiple images

### ISSUE THREAD

**Issue by Cloud9InfoIndia** [xceedsoftware/DocX#366]
Hi,

I am using DocX library to generate a Docx file report using a predefined template with placeholders. while generating report I am replacing placeholder with either text value or image(s). Text is getting replaced properly. But I am not able to replace multiple images against the same placeholder. It just replaces one image. I am using reportDoc.ReplaceTextWithObject method and passing a picture object to it. But I want to pass a collection of pictures. How can I do that? How can i replace multiple images for a single placeholder?

Thanks in advance.

**Comment by XceedBoucherS** [comment]
Hi,
This is currently not possible. When using document.ReplaceTextWithObject(), you can replace the text with a Picture, an Hyperlink or a Table. From what I understand, you would like to replace the text with many pictures. Can you give more details ?
How would the picture lines up ? How many do you plan to have in one replacement ? Will they be small ? This could modify the initial document layout. Can you submit a sample so we can evaluate this possibility.
Thank you.

**Comment by Cloud9InfoIndia** [comment]
Yes, I want to replace a placeholder with multiple images. Like If I have a placeholder {HousePictures} it should get replaced by all the images taken for the house (Ex: entrance pic, dining area pic, garden pic etc.). all these images should appear one after another and size is 200 by 200 dimensions. with this size It should show/arrange two images in one line. Images collection can have max 10 images. I have an ImageList collection and I am looping through it to convert each image into Picture object and then replacing the placeholder with picture object but this way it overwrites all the images except the last one.
Is there any other way with which I can achieve this?

**Comment by XceedBoucherS** [comment]
Currently, you would need as many placeholders as images you want to replace. Calling document.ReplaceTextWithObject() for every placeholder you have would replace them wit a specific image.

What if you have a document with many placeholders (let's say 10, from {HousePictures1} to {HousePictures10}). Then you parse your pictures and replaces the first placeholders in the document. If your document has more placeholders than pictures, you could always call a document.ReplaceText() to replace  {HousePicturesX} with empty strings. Your final document would end up with only the number of pictures you have.
